core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -314,6 +315,34 @@ public static String getClientHostName() {
     return getLocalHostName();
   }
 
+  /**
+   * Gets a local node name from configuration if it is available, falling back on localhost lookup.
+   *
+   * @return the local node name
+   */
+  public static synchronized String getLocalNodeName() {
+    switch (CommonUtils.PROCESS_TYPE.get()) {
+      case CLIENT:
+        if (Configuration.containsKey(PropertyKey.USER_HOSTNAME)) {
+          return Configuration.get(PropertyKey.USER_HOSTNAME);
+        }
+        break;
+      case MASTER:
+        if (Configuration.containsKey(PropertyKey.MASTER_HOSTNAME)) {
+          return Configuration.get(PropertyKey.MASTER_HOSTNAME);
+        }
+        break;
+      case WORKER:
+        if (Configuration.containsKey(PropertyKey.WORKER_HOSTNAME)) {
+          return Configuration.get(PropertyKey.WORKER_HOSTNAME);
+        }
+        break;
+      default:
```

Comment:
I don't think we currently have java code that talks to the proxy. If the proxy wants to use locality information, it should use tiered locality configuration instead of using PROXY_HOSTNAME.

This method is only used as a fallback when the user doesn't explicitly configure tiered locality.